<a href="https://colab.research.google.com/github/ankita130698/Text-to-SQL/blob/main/Simplified.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from transformers import pipeline

# Load pre-trained BERT model for sentiment analysis
sentiment_classifier = pipeline('sentiment-analysis', model='nlptown/bert-base-multilingual-uncased-sentiment')

# Sample text
texts = [
    "I love this product! It works wonders.",
    "I'm very disappointed with the service.",
    "This is the best day of my life!",
    "I'm so frustrated with the delays."
]

# Detect sentiments
for text in texts:
    sentiment = sentiment_classifier(text)
    print(f"Text: {text}\nSentiment: {sentiment}\n")


config.json:   0%|          | 0.00/953 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/669M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/39.0 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/872k [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

Text: I love this product! It works wonders.
Sentiment: [{'label': '5 stars', 'score': 0.9001321792602539}]

Text: I'm very disappointed with the service.
Sentiment: [{'label': '2 stars', 'score': 0.5437463521957397}]

Text: This is the best day of my life!
Sentiment: [{'label': '5 stars', 'score': 0.9460911750793457}]

Text: I'm so frustrated with the delays.
Sentiment: [{'label': '2 stars', 'score': 0.5003401637077332}]



The outputs of the sentiment analysis model provide insight into how text can be classified into different sentiment categories. Here, the nlptown/bert-base-multilingual-uncased-sentiment model was used, which classifies text into sentiment categories ranging from 1 to 5 stars. These categories can be interpreted as follows:

1 star: Very negative sentiment

2 stars: Negative sentiment

3 stars: Neutral sentiment

4 stars: Positive sentiment

5 stars: Very positive sentiment

**Text**: "I love this product! It works wonders."

**Sentiment**: [{'label': '5 stars', 'score': 0.9001321792602539}]

**Explanation**: The model classified this text as having a very positive sentiment (5 stars) with a high confidence score of 0.90. The use of positive words like "love" and "wonders" strongly indicates positive emotion.

In [1]:
from transformers import pipeline

# Load pre-trained BERT model for sentiment analysis
sentiment_classifier = pipeline('sentiment-analysis', model='nlptown/bert-base-multilingual-uncased-sentiment')

# Sample text
texts = [
    "I love this product! It works wonders.",
    "I'm very disappointed with the service.",
    "This is the best day of my life!",
    "I'm so frustrated with the delays.",
    "Oh great, another delay. Just what I needed.",
    "Fantastic! Another bug in the system."
]

# Function to detect sarcasm
def detect_sarcasm(text):
    sarcastic_indicators = [
        "Oh great",
        "Just what I needed",
        "Fantastic",
        "Yeah right",
        "As if"
    ]
    for indicator in sarcastic_indicators:
        if indicator.lower() in text.lower():
            return True
    return False

# Detect sentiments and sarcasm
for text in texts:
    sentiment = sentiment_classifier(text)
    is_sarcastic = detect_sarcasm(text)
    sarcasm_note = " (Sarcasm detected)" if is_sarcastic else ""
    print(f"Text: {text}\nSentiment: {sentiment}{sarcasm_note}\n")

/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:89: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/953 [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/669M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/39.0 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/872k [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

Text: I love this product! It works wonders.
Sentiment: [{'label': '5 stars', 'score': 0.9001321792602539}]

Text: I'm very disappointed with the service.
Sentiment: [{'label': '2 stars', 'score': 0.5437463521957397}]

Text: This is the best day of my life!
Sentiment: [{'label': '5 stars', 'score': 0.9460911750793457}]

Text: I'm so frustrated with the delays.
Sentiment: [{'label': '2 stars', 'score': 0.5003401637077332}]

Text: Oh great, another delay. Just what I needed.
Sentiment: [{'label': '5 stars', 'score': 0.6787647008895874}] (Sarcasm detected)

Text: Fantastic! Another bug in the system.
Sentiment: [{'label': '5 stars', 'score': 0.6442019939422607}] (Sarcasm detected)

